### PR TITLE
setting for new level progression bars

### PR DESCRIPTION
**Pull request by jatloe**
image
image

### CODE REVIEW

File: rpg/levelsys.py [hunk 1]
```diff
@@ -48,22 +48,33 @@ def __init__(
...
     @property
     def level(self) -> int:
         return int(max(0, self.value) ** 0.5)
 
     @property
     def percent(self) -> int:
-        eps = 0.40  # Make sure text fits in the bar
-        k = (self.value + eps * self.max_value) / ((1 + eps) * self.max_value)
+        eps = 0.4  # Make sure text fits in the bar
+        if not self.new_progress:
```
Comment (Incompleteusern) [inline rpg/levelsys.py:68]: Invert this if statement

File: rpg/levelsys.py [hunk 2]
```diff
@@ -48,22 +48,33 @@ def __init__(
...
+            k = (self.value + eps * self.max_value) / ((1 + eps) * self.max_value)
+        else:
+            eps = 0.25  # Make progress more visually clear
+            lvl = self.level
+            prev_value = lvl**2
+            next_value = (lvl + 1) ** 2
+            k = ((self.value - prev_value) + eps * (next_value - prev_value)) / (
+                (1 + eps) * (next_value - prev_value)
+            )
```
Comment (Incompleteusern) [inline rpg/levelsys.py:77]: Perhaps simplify as 
```python
            k = ((self.value - prev_value) / (next_value - prev_value) + eps) / (1 + eps)
```

File: core/migrations/0029_userprofile.py [hunk 3]
```diff
@@ -48,6 +48,14 @@ class Migration(migrations.Migration):
                         verbose_name="Show locked",
                     ),
                 ),
+                (
+                    "new_progress",
```
Comment (Incompleteusern) [inline core/migrations/0029_userprofile.py:52]: I'd rename this from `new_progress` to like `next_progress` or something. IIRC Evan suggests that this is actually older than the current system.

**Comment by jatloe** [comment]
bruh what ok I will change it to 144 locally

**Comment by jatloe** [comment]
help

**Comment by jatloe** [comment]
rawr

**Comment by vEnhance** [comment]
what does `np` stand for

**Comment by jatloe** [comment]
it was supposed to stand for new_progress but then I changed it to dynamic_progress and never changed np

**Comment by jatloe** [comment]
why doesn't the type check work? it works locally

**Comment by vEnhance** [comment]
can we make the variable name consistent (and maybe longer, for readability)

also the default value in `core/models.py`

re typecheck: pyright version?

**Comment by vEnhance** [comment]
oh it's not typecheck, if you read the log something got borked in the migration step

**Comment by jatloe** [comment]
lets try this again!!!!!!!!!!!!!!

**Comment by jatloe** [comment]
please?

**Comment by jatloe** [comment]
YAY

**Pull request merged**